source/fundamentals/typescript.txt

```diff
@@ -122,6 +121,45 @@ The type error raised by the preceding code snippet is as follows:
    ...
    Type '"Sylvilagus"' is not assignable to type 'Condition<"Canis" | "Felis">'.
 
+Creating Keys from Variables
+````````````````````````````
+
+If you need to query a collection or perform another operation using a
+key created from specified variables, you must use an ``as const``
+assertion when specifying the key. This mechanism will allow your
+code to compile successfully as long as the input types are correct.
+
+The following code snippet defines the ``ClassificationPet`` interface
+and the ``Mealtime`` interface. ``ClassificationPet`` includes a
+``mealtimes`` field that contains an array of ``Mealtime`` interfaces,
+each of which includes a ``time`` field:
+
+.. code-block:: typescript
+
+   interface ClassificationPet {
+     name: string;
+     mealtimes: Mealtime[];
+   }
+   
+   interface Mealtime{
+     time: string;
+     amount: number;
+   }
+
+The following code snippet performs a find and update operation on a
```

Comment:
[Compound modifier](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#hyphens-in-compound-modifiers)

```suggestion
The following code snippet performs a find-and-update operation on a
```